src/components/notification/popup/index.tsx

```diff
@@ -4,16 +4,42 @@ import { NotificationListContainer } from '../list/container';
 
 import './style.scss';
 
-export interface Properties {}
+export interface Properties {
+  onClickOutside: () => void;
+}
 
 export class NotificationPopup extends React.Component<Properties> {
+  private notificationPopupRef;
+
+  constructor(props) {
+    super(props);
+    this.notificationPopupRef = React.createRef();
+  }
+
   render() {
     return this.renderPopup();
   }
 
+  componentDidMount() {
+    document.addEventListener('mousedown', this.clickOutsideNotificationPopup, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.clickOutsideNotificationPopup, false);
```

Comment:
I think moving away after you pressed your mouse it won't cancel the click en general, it will cancel the click only in a particular pixel (dom element). in our case even you press down your mouse outside the notification popup and you move away you cannot know that you have canceled the click to close the notification and it will trigger mousedown|up|click event.

Reply:
It prevents firing a click event in the area that you did the mousedown. I do it all the time.